### DISCUSSION TITLE
Event when a particular Node type is added to the editor

### DISCUSSION THREAD

**Discussion by dominictobias-bullish** [facebook/lexical#6800] [Q&A] [answered]
How can I listen for when my custom node is added to the editor (`FileAttachmentNode`)?

I'm hoping there's something more efficient than the `OnChangePlugin` which would require me to recurse over all the nodes and check.

**Comment by etrepum** [comment, marked as answer]
Mutation listener

**Reply by dominictobias-bullish** [reply]
Thanks! `editor.registerMutationListener(FileAttachmentNode, nodeMutations => {...}`
